File: internal/api/dto/response/list.go
package response

// ListResponse is a generic response for list endpoints
// swagger:response listResponse
type ListResponse struct {
	Data interface{} `json:"data"`
	Meta Meta        `json:"meta"`
}

type Meta struct {
	Total int `json:"total"`
	Page  int `json:"page"`
	Limit int `json:"limit"`
}

File: web/apps/web/lib/schemas/list.ts
import {z} from "zod"

export const ListResponseSchema = z.object({
  data: z.array(z.any()),
  meta: z.object({
    total: z.number(),
    page: z.number(),
    limit: z.number(),
  })
})
export type  ListResponse = z.infer<typeof ListResponseSchema>
